packages/core/js-client/src/jsPeer/FluencePeer.ts

```diff
@@ -303,47 +325,193 @@ export abstract class FluencePeer {
                     log_particle.trace('id %s. call results: %j', item.particle.id, item.callResults);
                 }),
                 filterExpiredParticles(this._expireParticle.bind(this)),
-            )
-            .subscribe((item) => {
-                const p = item.particle;
-                let particlesQueue = this._particleQueues.get(p.id);
-
-                if (!particlesQueue) {
-                    particlesQueue = this._createParticlesProcessingQueue();
-                    this._particleQueues.set(p.id, particlesQueue);
-
-                    const timeout = setTimeout(() => {
-                        this._expireParticle(item);
-                    }, getActualTTL(p));
-
-                    this._timeouts.push(timeout);
-                }
-
-                particlesQueue.next(item);
-            });
-
-        this._outgoingParticles.subscribe((item) => {
-            // Do not send particle after the peer has been stopped
-            if (!this.isInitialized) {
-                return;
-            }
+                groupBy(item => item.particle.id),
```

Comment:
Interesting... Does it mean that there are multiple particles with the same id at the same time?

Reply:
Particles with the same id are grouped and processed one by one